core/domain/user_query_jobs_one_off.py

```diff
@@ -102,3 +102,13 @@ def reduce(query_model_id, stringified_user_ids):
         user_ids = [ast.literal_eval(v) for v in stringified_user_ids]
         query_model.user_ids = [str(user_id) for user_id in user_ids]
         query_model.put()
+
+    @classmethod
+    def _post_completed_hook(cls, job_id):
+        # TODO: change query status in query model to completed.
```

Comment:
Hi @seanlip, this is actually part of milestone 4 (Implement success and failure) conditions but I have added them here for few difficulties I have been facing. Let me describe issue:

whenever a query has been completed, we have to change its status from processing to completed or failed. This can not be done inside `reduce()` function because of 2 reasons:
1. If no user is qualified for given query then reduce won't be called at all.
2. If query failed during map procedure reduce won't be called.

So I have to override _post_complete and _post_failure methods. To access query model from these functions we need query id, which is not available. Only job id is available. Now we need some way around this issue so that we can have query id in these methods. Then and only then completion and failure conditions can be performed. (That's why if you would run query from browser right now, it will never show completed status. It will always stay at processing status.)

Now to tackle this I can think of 2 ways.
1. We store additional params in job model for each job.
2. We store job id in query model along side query id.

So what do you think about this issue ? 
I added these methods here to ask this question and to tell that in this PR query will never go to complete status.

Thanks